src/pages/home/report/EmojiPickerMenu/index.js

```diff
@@ -0,0 +1,132 @@
+        // If this number is changed, emojis.js will need to be updated to have the proper number of spacer elements
+        // around each header.
+        this.numColumns = 8;
+
+        // This is the indices of each category of emojis
+        // The positions are static, and are calculated as index/numColumns (8 in our case)
+        // This is because each row of 8 emojis counts as one index
+        // If this emojis are ever added to emojis.js this will need to be updated or things will break
+        this.unfilteredHeaderIndices = [0, 34, 60, 88, 99, 121, 148];
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 300, false);
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: this.unfilteredHeaderIndices,
+        };
+    }
+
+    /**
+     * Filter the entire list of emojis to only emojis that have the search term in their keywords
+     *
+     * @param {String} searchTerm
+     */
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            // There are no headers when searching, so we need to re-make them sticky when there is no search term
+            this.setState({filteredEmojis: emojis, headerIndices: this.unfilteredHeaderIndices});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        _.each(emojis, (emoji) => {
+            if (emoji.header || emoji.code === CONST.EMOJI_SPACER) {
+                return;
+            }
+
+            if (_.find(emoji.keywords, keyword => keyword.includes(normalizedSearchTerm))) {
+                newFilteredEmojiList.push(emoji);
+            }
+        });
+
+        // Remove sticky header indices. There are no headers while searching and we don't want to make emojis sticky
+        this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: []});
+    }
+
+    /**
+     * Given an emoji item object, render a component based on its type.
+     * Items with the code "SPACER" return nothing and are used to fill rows up to 8
+     * so that the sticky headers function properly
+     *
+     * @param {Object} item
+     * @returns {*}
+     */
+    renderItem({item}) {
+        if (item.code === CONST.EMOJI_SPACER) {
+            return null;
+        }
+
+        if (item.header) {
+            return (
+                <Text style={styles.emojiHeaderStyle}>
+                    {item.code}
+                </Text>
+            );
+        }
+
+        return (
+            <EmojiPickerMenuItem
+                onPress={this.props.onEmojiSelected}
+                emoji={item.code}
+            />
+        );
+    }
+
+    render() {
+        return (
+            <View style={styles.emojiPickerContainer}>
+                <View style={[styles.pt4, styles.ph4, styles.pb1]}>
```

Comment:
Just thinking out loud (and we don't need to do this in this PR), but what if we made utility functions for margin and padding that worked sort of like the shorthand css padding property:

```
// Provide padding for the top, right, bottom, left; respectively (using the same increments of 4 that our "bootstrap" styles use)
styles.padding(4, 4, 1, 4);
```